### PR TITLE
Add Google OAuth Login (OpenID Connect) support

### PR DESCRIPTION
**Pull request by sleeper-service**
Added a clone of com.epimorphics.registry.webapi.ProcessOpenId to provide ProcessOAuth2, changing the OpenID specific methods to OAuth/OpenID Connect. using the Apache Oltu project client library
 
Redirected the targets of  the "Login with google account" and "Register using google account" buttons to invoke the relevant methods of ProcessOAuth2.
 
Added a caching http client for retrieving the google OAuth2 discovery document and a new configuration file /opt/ldregistry/config/oauth.conf containing three settings:

ldregistry.client-id  = <google oauth clients id>
ldregistry.client-secret = <client secret>
ldregistry.usehttps = <true|false> 

usehttps forces redirection to https for requests after successful authentication and back to https after logout when https is not being used by default for all requests.

### CODE REVIEW

**Comment by marqh** [comment]
excellent stuff sleeper-service

I have pushed this branch to one of my development instances, together with an SSL certificate and I can confirm that the Google Oauth authentication is working effectively

File: pom.xml [hunk 1]
```diff
@@ -1,398 +1,427 @@
...
-  <version>0.0.6-SNAPSHOT</version>
-  <name>Registry Core</name>
-  <url>http://maven.apache.org</url>
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.github.ukgovld</groupId>
+    <artifactId>registry-core</artifactId>
+    <packaging>war</packaging>
+    <version>0.0.6-SNAPSHOT</version>
+    <name>Registry Core</name>
+    <url>http://maven.apache.org</url>
```
Comment (marqh) [inline pom.xml]: these are not content changes, just formatting.  I have put a PR against this PR to fix them
https://github.com/sleeper-service/registry-core/pull/2

this should make it easier to see the functional changes to the pom.xml

**Comment by der** [comment]
Sounds great, many thanks for this. Will attempt to review and merge as soon as I can though pressure of other work means that won't be immediate.

**Comment by marqh** [comment]
I have been through this code in detail and I am very pleased with the capabilities and the implementation.

I think there are considerations about integration and extensibility of the user interface, but i would like to see them addressed in future work, rather than trying to handle all of those thought before adopting this code

I highly recommend that we adopt this PR as it is now.

**Comment by der** [comment]
Sorry to have been so slow to look at this, pressure of other deadlines.

Changes looks great and after a few minor issues I've verified this works for me and merged the PR into master.

1. There were a few "yellow flags" (unused variables etc) that I've cleaned up in doing the merge.

2. A couple of the new files have a copyright header of Newcastle University but with an Apache 2 licence, that's fine, but others have no copyright/licence header at all. Would it be possible to add suitable headers and do a new PR for those?

3. It would be useful to add a some documentation on how to set up to use this. I'll sketch something and ask you both to review/update.

Many thanks for contributing this.

**Comment by der** [comment]
4. The dependency on Otlu snapshot is a little awkward. Makes it hard to a released version of registry-core. I assume the 1.0.0 released version had some problem which made it necessary to use the snapshot?

**Pull request merged**